examples/compositor.c

```diff
@@ -83,6 +85,10 @@ static void handle_output_frame(struct output_state *output, struct timespec *ts
 	wl_list_for_each(xdg_surface, &sample->xdg_shell->surfaces, link) {
 		output_frame_handle_surface(sample, wlr_output, ts, xdg_surface->surface);
 	}
+	struct wlr_x11_window *x11_window;
+	wl_list_for_each(x11_window, &sample->xwayland->displayable_windows, link) {
+		output_frame_handle_surface(sample, wlr_output, ts, x11_window->surface);
+	}
```

Comment:
Per the spec we should only be rendering views that have a role assigned to them. Each role may also need to be handled subtly differently. Iterating over each of the containers of surfaces with roles attached also separates concerns better.